src/dnshelper.cpp

```diff
@@ -120,19 +120,10 @@ bool DNSHelper::shouldExcludeDNS() {
 
   bool isLocalDNS =
       RFC1918::contains(dnsAddress) || RFC4193::contains(dnsAddress);
-  if (!Feature::get(Feature::Feature_lanAccess)->isSupported() && isLocalDNS) {
-    // In case we cant use lan access, we must exclude it (the platform already
-    // does the magic for us).
-    return false;
-  }
-
-  if (isLocalDNS && settings->localNetworkAccess()) {
+  if (isLocalDNS) {
     // DNS is lan, but we already excluded local-ip's, all good.
     return false;
   }
 
-  if (!isLocalDNS) {
-    return false;
-  }
   return true;
```

Comment:
I'm talking about this. If we are here, the IP address is not part of LAN. And we should not exclude it. This should be `return false`. And at this point we can remove this method entirely.